File: Triatlon-Java/TriatlonPersistance/src/main/java/repository/databases/InscriereDB.java
package repository.databases;

import domain.*;
import repository.Utils.JdbcUtils;
import repository.interfaces.InscriereRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class InscriereDB implements InscriereRepository{
    private final JdbcUtils jdbcUtils;

    private static final Logger logger = LogManager.getLogger();

    public InscriereDB(Properties props) {
        logger.info("Initializing InscriereDB with properties: {} ", props);
        jdbcUtils = new JdbcUtils(props);
    }
    @Override
    public List<Long> findParticipantsByProba(Long idProba) {
        logger.traceEntry("finding participants by proba with {}", idProba);
        Connection con = jdbcUtils.getConnection();
        List<Long> participanti = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from inscriere where id_proba= ?")) {
            preStmt.setLong(1, idProba);
            try(ResultSet result = preStmt.executeQuery()) {
                while(result.next()) {
                    Long id = result.getLong("id_participant");
                    participanti.add(id);
                }
            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(participanti);
        return participanti;
    }

    @Override
    public List<Inscriere> findInscriereByProbaAndParticipant(Long idProba, Long idParticipant) {
        logger.traceEntry("finding inscriere by proba and participant with {}", idProba, idParticipant);
        Connection con = jdbcUtils.getConnection();
        List<Inscriere> inscrieri = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from inscriere where id_proba= ? and id_participant= ?")) {
            preStmt.setLong(1, idProba);
            preStmt.setLong(2, idParticipant);
            try(ResultSet result = preStmt.executeQuery()) {
                while(result.next()) {
                    Long id = result.getLong("id");
                    Long punctaj = result.getLong("punctaj");
                    Inscriere inscriere = new Inscriere(idParticipant, idProba, punctaj);
                    inscriere.setId(id);
                    inscrieri.add(inscriere);
                }
            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(inscrieri);
        return inscrieri;
    }

    @Override
    public Inscriere findOne(Long aLong) {
        logger.traceEntry("finding inscriere with {}", aLong);
        Connection con = jdbcUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("select * from inscriere where id= ?")) {
            preStmt.setLong(1, aLong);
            try(ResultSet result = preStmt.executeQuery()) {
                if(result.next()) {
                    Long id = result.getLong("id");
                    Long idParticipant = result.getLong("id_participant");
                    Long idProba = result.getLong("id_proba");
                    Long punctaj = result.getLong("punctaj");
                    Inscriere inscriere = new Inscriere(idParticipant, idProba, punctaj);
                    inscriere.setId(id);
                    logger.traceExit(inscriere);
                    return inscriere;
                }
            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit("No inscriere found with id {}", aLong);
        return null;
    }

    @Override
    public Iterable<Inscriere> findAll() {
        logger.traceEntry();
        Connection con = jdbcUtils.getConnection();
        List<Inscriere> inscrieri = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from inscriere")) {
            try(ResultSet result = preStmt.executeQuery()) {
                while(result.next()) {
                    Long id = result.getLong("id");
                    Long idParticipant = result.getLong("id_participant");
                    Long idProba = result.getLong("id_proba");
                    Long punctaj = result.getLong("punctaj");
                    Inscriere inscriere = new Inscriere(idParticipant, idProba, punctaj);
                    inscriere.setId(id);
                    inscrieri.add(inscriere);
                }
            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(inscrieri);
        return inscrieri;
    }

    @Override
    public Inscriere save(Inscriere entity) {
        logger.traceEntry("saving inscriere {} ", entity);
        Connection con = jdbcUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("insert into inscriere(id_participant, id_proba, punctaj) values (?,?,?)")) {
            preStmt.setLong(1, entity.getIdParticipant());
            preStmt.setLong(2, entity.getIdProba());
            preStmt.setLong(3, entity.getPunctaj());
            int result = preStmt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(entity);
        return entity;
    }

    @Override
    public Inscriere delete(Long aLong) {
        return null;
    }

    @Override
    public Inscriere update(Inscriere entity1, Inscriere entity2) {
        logger.traceEntry("updating inscriere {} ", entity1);
        Connection con = jdbcUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("update inscriere set punctaj=? where id_participant=? and id_proba=?")) {
            preStmt.setLong(1, entity2.getPunctaj());
            preStmt.setLong(2, entity2.getIdParticipant());
            preStmt.setLong(3, entity2.getIdProba());
            int result = preStmt.executeUpdate();
            logger.trace("Updated {} instances", result);
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(entity2);
        return entity2;
    }


}

File: Triatlon-C#/TriatlonPersistance/databases/InscriereDB.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Data.SQLite;
using System.Windows.Forms;
using ConcursTriatlon.repository.interfaces;
using log4net;
using TriatlonModel;

namespace ConcursTriatlon.repository.databases
{
    public class InscriereDB : InscriereRepository
    {
        public static readonly ILog Log = LogManager.GetLogger("InscriereDB");
        private string connectionString;
        
        public InscriereDB(string connectionString)
        {
            Log.Info("Creating InscriereDB");
            this.connectionString = connectionString;
        }
        
        public List<long> findParticipantByProba(long idProba)
        {
            Log.InfoFormat("Entering findParticipantByProba with value {0}", idProba);
            List<long> idParticipanti = new List<long>();
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select id_participant from inscriere where id_proba=@idProba", connection);
                    command.Parameters.AddWithValue("@idProba", idProba);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long idParticipant = reader.GetInt64(0);
                        idParticipanti.Add(idParticipant);
                    }
                    reader.Close();

                }
                catch (Exception e)
                {
                    Log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            Log.InfoFormat("Exiting findParticipantByProba with value {0}", idParticipanti);
            return idParticipanti;
        }

        public List<long> findInscriereByProbaAndParticipant(long idProba, long idParticipant)
        {
            Log.InfoFormat("Entering findInscriereByProbaAndParticipant with value {0} and {1}", idProba, idParticipant);
            List<long> idInscrieri = new List<long>();
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select id from inscriere where id_proba=@idProba and id_participant=@idParticipant", connection);
                    command.Parameters.AddWithValue("@idProba", idProba);
                    command.Parameters.AddWithValue("@idParticipant", idParticipant);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long idInscriere = reader.GetInt64(0);
                        idInscrieri.Add(idInscriere);
                    }
                    reader.Close();

                }
                catch (Exception e)
                {
                    Log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            Log.InfoFormat("Exiting findInscriereByProbaAndParticipant with value {0}", idInscrieri);
            return idInscrieri;
        }
        
        public Inscriere findOne(long id)
        {
            Log.InfoFormat("Entering findOne with value {0}", id);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select * from inscriere where id=@id", connection);
                    command.Parameters.AddWithValue("@id", id);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long idProba = reader.GetInt64(1);
                        long idParticipant = reader.GetInt64(2);
                        Inscriere inscriere = new Inscriere(id, idProba, idParticipant);
                        Log.InfoFormat("Exiting findOne with value {0}", inscriere);
                        reader.Close();

                        return inscriere;
                    }
                } catch (Exception e)
                {
                    Log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            Log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public IEnumerable<Inscriere> findAll()
        {
            Log.InfoFormat("Entering findAll");
            List<Inscriere> inscrieri = new List<Inscriere>();
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select * from inscriere", connection);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long id = reader.GetInt64(0);
                        long idParticipant = reader.GetInt64(1);
                        long idProba = reader.GetInt64(2);
                        long punctaj = reader.GetInt64(3);
                        Inscriere inscriere = new Inscriere(idParticipant, idProba, punctaj);
                        inscriere.setId(id);
                        inscrieri.Add(inscriere);
                    }
                    reader.Close();

                }
                catch (Exception e)
                {
                    Log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
                //connection.Close();
            }
            
            Log.InfoFormat("Exiting findAll with value {0}", inscrieri);
            return inscrieri;
        }

        public Inscriere save(Inscriere entity)
        {
            Log.InfoFormat("Entering save with value {0}", entity);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("INSERT INTO inscriere (id_participant, id_proba, punctaj) values (@idParticipant, @idProba, @punctaj)", connection);
                    command.Parameters.AddWithValue("@idParticipant", entity.IdParticipant);
                    command.Parameters.AddWithValue("@idProba", entity.IdProba);
                    command.Parameters.AddWithValue("@punctaj", entity.Punctaj);

                    MessageBox.Show("Inscriere IN PROCES!");
                    MessageBox.Show(entity.ToString());
                    command.ExecuteNonQuery();
                    MessageBox.Show("Inscriere REUSITA!");
                } catch (SQLiteException e)
                {
                    Log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
                
                catch (Exception e)
                {
                    Log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            Log.InfoFormat("Exiting save with value {0}", entity);
            return entity;
        }

        public Inscriere delete(long id)
        {
            return null;
        }

        public Inscriere update(Inscriere entity1, Inscriere entity2)
        {
            Log.InfoFormat("Entering update with value {0} and {1}", entity1, entity2);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("update inscriere set punctaj=@punctaj where id_participant=@idParticipant and id_proba=@idProba", connection);
                    command.Parameters.AddWithValue("@punctaj", entity2.Punctaj);
                    command.Parameters.AddWithValue("@idParticipant", entity2.IdParticipant);
                    command.Parameters.AddWithValue("@idProba", entity2.IdProba);
                    command.ExecuteNonQuery();
                } catch (Exception e)
                {
                    Log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            Log.InfoFormat("Exiting update with value {0}", entity2);
            return entity2;
        }
        
    }
}

